redwood-treehouse-lazylayout-uiview/src/commonMain/kotlin/app/cash/redwood/treehouse/lazylayout/uiview/UIViewLazyList.kt

```diff
@@ -40,50 +50,76 @@ internal class UIViewLazyList<A : AppService>(
   treehouseApp: TreehouseApp<A>,
   widgetSystem: WidgetSystem,
 ) : LazyList<UIView> {
-  private val dataSource = TableViewDataSource(treehouseApp, widgetSystem)
-  private val root = UITableView().apply {
-    this.dataSource = this@UIViewLazyList.dataSource
-  }
+  private val layout = UICollectionViewFlowLayout()
+  private val viewController = CollectionViewController(treehouseApp, widgetSystem, layout)
 
   override fun isVertical(isVertical: Boolean) {
-    if (!isVertical) {
-      // TODO UITableView only supports vertical scrolling. Switch to UICollectionView.
-      TODO()
+    layout.scrollDirection = if (isVertical) {
+      UICollectionViewScrollDirectionVertical
+    } else {
+      UICollectionViewScrollDirectionHorizontal
     }
   }
 
   override fun intervals(intervals: List<LazyListInterval>) {
-    dataSource.intervals = intervals
-    root.reloadData()
+    viewController.dataSource.intervals = intervals
+    viewController.collectionView.reloadData()
   }
 
   override var layoutModifiers: LayoutModifier = LayoutModifier
 
-  override val value: UIView get() = root
+  override val value: UIView get() = viewController.collectionView
+}
+
+private class CollectionViewController<A : AppService>(
```

Comment:
When the `UICollectionView` is vertically scrolling, the search bar is pushed off the screen. This might be related to https://github.com/cashapp/redwood/issues/770? Or should I be constraining the height somehow?

Reply:
In which direction is it pushed off the screen? Would you mind uploading a screen recording?

Reply:
Sorry, it's not super clear from the description. Here's a couple of vids — one showcasing `LazyColumn` and the other `LazyRow`.

https://user-images.githubusercontent.com/6900601/236233945-f386b339-6c2c-4460-a038-0f3ab10d11a6.mp4

https://user-images.githubusercontent.com/6900601/236233958-6c858229-0ad6-422e-91b8-c2e4d8ba1c57.mp4

Reply:
Ah okay! After Googling around a bit, people mentioned using `NSLayoutConstraint.activateConstraints`, but I've tried using that with no luck. Is that the correct API to be looking in to?

Reply:
Have you tried calling [sizeToFit](https://developer.apple.com/documentation/uikit/uiview/1622630-sizetofit)?

Reply:
What size is the collection view when you use the view debugger. The constraints are only applicable if we're using autolayout, so probably not. 

I'll pull the branch today and poke around. 

I'm guessing it's the `Column` view that this lazy view is contained in that isn't constraining it's size. Do you know which `Column` implementation it's using? There's still bugs in the FlexContainer and Yoga implementations

Reply:
> The constraints are only applicable if we're using autolayout, so probably not.

Ah, got it.

> I'm guessing it's the `Column` view that this lazy view is contained in that isn't constraining it's size. Do you know which `Column` implementation it's using? There's still bugs in the FlexContainer and Yoga implementations

Okay, if it's a known issue then we could just merge this as-is then.

> I'll pull the branch today and poke around.

That'd be sweet — thank you!